### PR TITLE
Support Complete task user task action in Zeebe client

### LINKED ISSUE #16447: Support `Complete task` user task action
## Description

* Based on the result of https://github.com/camunda/zeebe/issues/16184, we introduce a "complete" method for user tasks in the Zeebe Java client.
* The method requires a `userTaskKey` and optionally takes an `action` of type `String` and `variables` of type `Map<String, Object`.
* The method completes a user task using Zeebe's REST endpoint `POST /v1/user-tasks/{userTaskKey}/completion` created with https://github.com/camunda/zeebe/issues/16118.

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

* Add user task completion support to the Zeebe Java client via `ZeebeClient#newUserTaskCompleteCommand`.
* The command takes a user task key and can optionally take an `action` and a `Map<String, Object` of `variables`.

**Note**: due to the completion REST request object expecting a proper Map for variables, we currently only support a proper Map for the client command as well. This might be extended by follow-up issues.

## Related issues

closes #16447 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [x] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by tmetzke** [comment]
I have a local QA test that I won't submit for now since it copies a lot of the helper methods from GrpcClientRule to easily create example data. But it works nicely except for the 404 rejection that runs into an NPE as mentioned [over here](https://github.com/camunda/zeebe/pull/16592#discussion_r1508541953).

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java
Comment (koevskinikola) [inline zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java:473]: :thinking: Should we also note here that this command only supports communication over REST?

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CompleteUserTaskCommandImpl.java
Comment (koevskinikola) [inline zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CompleteUserTaskCommandImpl.java:34]: :+1: Nice!

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java
Comment (koevskinikola) [inline zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java:105]: :thought_balloon: I guess we'll refactor this method once we implement HTTP POST endpoints that return some content in the response.

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Looks great @tmetzke. There's some minor feedback for the `requestTimeout` property. It would be good to fix it before merging.

I know that we tested the functionality manually, but it would be good if we can introduce some QA tests for this command. We can also do that after the `8.5-alpha2` release, but it would be good if we already have it. It's not a blocker for getting this PR merged.

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java
Comment (tmetzke) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java:450]: I agree 👍

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java
Comment (tmetzke) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java:473]: Yep, will add it

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CompleteUserTaskCommandImpl.java
Comment (tmetzke) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CompleteUserTaskCommandImpl.java:55]: Ah ok, yeah, got it. Will fix it for both.

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java
Comment (tmetzke) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java:105]: Yep, but we don't have any yet, so 🤷

**Comment by tmetzke** [comment]
> Looks great @tmetzke. There's some minor feedback for the `requestTimeout` property. It would be good to fix it before merging.
> 
> I know that we tested the functionality manually, but it would be good if we can introduce some QA tests for this command. We can also do that after the `8.5-alpha2` release, but it would be good if we already have it. It's not a blocker for getting this PR merged.

I won't get QA in here today, I think. Tooling is not ideal yet for that, at least that's what it feels like to me currently. If we still want to merge this to have it in the next alpha, we'd have to go without QA and deliver that after the alpha.

**Comment by tmetzke** [comment]
Regarding failing `TopologyClusterTest`, I don't see how the changes here could have introduced that. It fails reliably though but could still simply be very flaky. It didn't fail on `main` so far.

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java
Comment (tmetzke) [inline zeebe/clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java:476]:
```suggestion
   * <p>This command is only sent via REST over HTTP, not via gRPC <br>
```

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java
Comment (tmetzke) [inline zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java:116]: @koevskinikola, this is the issue, I would say. We're always sending a POST 🙈

Comment (koevskinikola) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java]: @ana-vinogradova-camunda I pushed a fix for the issue. Let me know if you can review the PR before we merge it.

**Review by koevskinikola — APPROVED** [review]
LGTM :tada:

**Review by ana-vinogradova-camunda — APPROVED** [review]
LGTM 🚀

**Comment by tmetzke** [comment]
@koevskinikola, the backport is obsolete now, correct? You backported it in a different way already?

**Pull request merged**
